### ISSUE TITLE
Feature: Reduce the size of recordings by elminating duplicate requests

### ISSUE THREAD

**Issue by theunrepentantgeek** [Azure/azure-service-operator#1891]
Our HTTP test recordings currently include a lot of noise where the operator is waiting for ARM to finish a prior request.

We could reduce the size of the recordings (and speed up playback testing) by collapsing sequences of identical requests into a single result.

Two examples come to mind:

1. While waiting for a new resource to be created, the operator issues repeated GET requests that all return 404, moving to the next stage only once a successful request is returned. This sequence could be replaced with a single GET/404.

2. While waiting for an existing resource to be deleted, the operator issues repeated GET requests that return 200 with (near) identical payloads. This sequence could be replaced with a single GET/200.

There are likely many other cases that we can identify.

**Comment by theunrepentantgeek** [comment]
This still seems nice to have.

**Comment by matthchr** [comment]
Still seems nice to have, but no rush

**Comment by theunrepentantgeek** [comment]
I tried this while working on an upgrade of **go-vcr** - not only does eliminating duplicate 404s have a minimal effect on most recordings (cutting out one or two requests at best), but their lack makes the playback of the test fail every time. 

The potential savings aren't worth debugging the test failures.
